core/src/main/scala/tech/beshu/ror/accesscontrol/utils/TaskRuleResultOps.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.utils
+
+import monix.eval.Task
+import tech.beshu.ror.accesscontrol.blocks.BlockContext
+import tech.beshu.ror.accesscontrol.blocks.rules.Rule.RuleResult
+
+implicit final class TaskRuleResultOps[B <: BlockContext](private val task: Task[RuleResult[B]]) extends AnyVal {
```

Comment:
Maybe instead of this, we could:
1. provide Functor and Monad instance for RuleResult
2. use monad transformers 

WDYT?

Reply:
I like it much better than type alias and extension methods that I posted above. But for now I'd prefer not to do it. We could add this type and use it in new methods in the interface, but then we would have 2 conventions (Task[Result] and ResultT[Task]). So if we do that, I think it should be a separate PR with refactor, that would change it globally, not only in a few places, as in this PR. 

In this PR a single helper method is IMO enough, we wouldn't use the full power of the new type anyway, without doing the refactor first.